katas/content/nonlocal_games/ghz_classical_strategy/Verification.qs

```diff
@@ -0,0 +1,56 @@
+namespace Kata.Verification {
+    open Microsoft.Quantum.Arrays;
+    open Microsoft.Quantum.Convert;
+    open Microsoft.Quantum.Logical;    
+    open Microsoft.Quantum.Random;
+
+    function WinCondition_Reference (rst : Bool[], abc : Bool[]) : Bool {
+        return (rst[0] or rst[1] or rst[2]) == Xor(Xor(abc[0], abc[1]), abc[2]);
+    }
+
+    // All possible starting bits (r, s and t) that the referee can give
+    // to Alice, Bob and Charlie.
+    function RefereeBits () : Bool[][] {
+        return [[false, false, false],
+                [true, true, false],
+                [false, true, true],
+                [true, false, true]];
+    }
+
+    operation PlayClassicalGHZ_Reference (strategies : (Bool => Bool)[], inputs : Bool[]) : Bool[] {
+        if Length(strategies) != 3 or Length(inputs) != 3 {
+            return [];
+        }
+        let r = inputs[0];
+        let s = inputs[1];
+        let t = inputs[2];
+        let a = strategies[0](r);
+        let b = strategies[1](s);
+        let c = strategies[2](t);
+        return [a, b, c];
+    }
+
+    @EntryPoint()
+    operation CheckSolution() : Bool {
+        let inputs = RefereeBits();
+        let strategies = [Kata.AliceClassical, Kata.BobClassical, Kata.CharlieClassical];
+
+        let iterations = 10000;
+        mutable wins = 0;
+        for _ in 0 .. iterations - 1 {
+            for bits in inputs {
+                let abc = PlayClassicalGHZ_Reference(strategies, bits);
+                if WinCondition_Reference(bits, abc) {
+                    set wins = wins + 1;
+                }
+            }
+        }
+        // The solution is correct if the players win 75% (3/4) of the time.        
+        if wins < iterations*Length(inputs)*3/4 {
```

Comment:
I got unlucky, and my correct solution failed one of the first few runs. (It passed every run after that, but one can get unlucky). Let's add a couple percent to tolerate randomness - the original kata accepted the solution if the win % was 73%.

Reply:
This is quite interesting...
I redesigned the verification to eliminate the randomness and to have a precise calculation. So, the correct solution MUST NOT fail.
Let's try the following: since I redesigned the win criteria, reduced the number of iterations, and added the 'wins' into the fail message - we can test again.
If "unlucky test" ever happens" - I'll change to a lower percentage to tolerate the randomness.
@tcNickolas, what do you think?

Reply:
You're right, this testing code is deterministic, so it shouldn't fail randomly... 

I spent quite a few minutes clicking on "Run" button on the correct solution just now, and one of the runs did give me something other than "correct", but I was clicking so fast, I only could register the fact of a longer two-line error message, not read the text itself, before the next click overwrote it again :-(

I think we can keep it as is now, but also keep an eye on any odd failures - our CI build, for example, will run a lot more times than manual testing will allow, if this starts failing there too, we'll know it's not limited to my machine

Reply:
@tcNickolas 
I saw this problem when I clicked fast on "Run" button, so I recorded the session on my phone.
Per the recording, the "Correct" line disappears, and nothing else is displayed (per my video).

Reply:
It could be something about clicking _fast_ then rather than a low probability thing... Let's leave it to CI to decide between the two :-)